plugin/caddyfile/fromlabels_test.go

```diff
@@ -0,0 +1,114 @@
+package caddyfile
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFromLabels_Grouping(t *testing.T) {
+	labels := map[string]string{
+		"caddy":               "localhost",
+		"caddy.group.a":       "value-a",
+		"caddy.group.b":       "value-b",
+		"caddy.group.group.a": "value-a",
+		"caddy.group.group.b": "value-b",
+	}
+
+	caddyfileBlock := FromLabels(labels, nil)
+
+	const expectedCaddyfile = "localhost {\n" +
+		"	group {\n" +
+		"		a value-a\n" +
+		"		b value-b\n" +
+		"		group {\n" +
+		"			a value-a\n" +
+		"			b value-b\n" +
+		"		}\n" +
+		"	}\n" +
+		"}\n"
```

Comment:
Couldn't you just write this with ` template strings? A lot less Go syntax soup that way (applies for all the expected strings)

Reply:
Yes. I tried several times with ``. Basically, I would like to use `` but keep indentation instead of moving everything to left.
Tried with this: https://github.com/lithammer/dedent
But it was always adding one extra line at top.

Reply:
You could also put the test data in files in a `testdata` dir, would be easier to maintain I think. I could help with doing that if you like

Reply:
Yes, sure, help is very welcome!
I think it's good somehow to keep test input and expected output close to each other.
Maybe moving input labels also into files would solve that...
Or implementing some deindentation of ` ` strings.

Reply:
For Caddy I took this sort of approach:

https://github.com/caddyserver/caddy/blob/master/caddytest/integration/caddyfile_adapt/log_roll_days.txt

Single files with a bunch of dashes as a split delimiter, first part is Caddyfile, 2nd part is JSON.

Could do the same but with labels first, Caddyfile 2nd.